File: AndroidLab/ch2/src/main/java/com/example/ch2/section1/Test1_1Activity.java
package com.example.ch2.section1;

import android.os.Bundle;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.ch2.R;

public class Test1_1Activity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);

        //test1...... 코드에서 직접 view 클래스 이용해서 화면 구성...
//        TextView nameView = new TextView(this);
//        nameView.setText("홍길동");
//
//        TextView emailView = new TextView(this);
//        emailView.setText("a@a.com");
//
//        //화면에 여러 뷰가 같이 나온다면.. 객체의 계층구조로 구성해서 출력해야 한다.
//        //디렉토리 같은 다른 객체를 add 시킬 뷰가 있어야 한다..
//        LinearLayout linearLayout = new LinearLayout(this);
//        linearLayout.setOrientation(LinearLayout.VERTICAL);
//        //사이즈 지정때문에 준비...
//        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
//                LinearLayout.LayoutParams.WRAP_CONTENT,
//                ViewGroup.LayoutParams.WRAP_CONTENT
//        );
//
//        //디렉토리에 추가..
//        linearLayout.addView(nameView, params);
//        linearLayout.addView(emailView, params);
//        //화면 출력.. 매개변수에 뷰 계층의 root 객체를 지정
//        setContentView(linearLayout);
//
//
//        ViewCompat.setOnApplyWindowInsetsListener(linearLayout, (v, insets) -> {
//            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
//            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
//            return insets;
//        });


        //layout xml 로 화면 구성..........................
        setContentView(R.layout.activity_test11);

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
}

File: AndroidLab2/ch2/src/main/java/com/example/ch2/section1/Test1_1Activity.kt
package com.example.ch2.section1

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.ch2.R
import com.example.ch2.databinding.ActivityTest11Binding

class Test1_1Activity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        val binding = ActivityTest11Binding.inflate(layoutInflater)
        setContentView(binding.root)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        binding.saveBtn.setOnClickListener {
            //preference 객체 획득..
            //PRIVATE 이외에 READABLE, WRITEABLE 등이 있지만 해당 앱에서만 사용하는 PRIVATE 권장.
            val sharedPref = getSharedPreferences("my_prefs", MODE_PRIVATE)
            sharedPref.edit().run {
                putString("data1", "hello")
                putInt("data2", 10)
                apply()
            }

            //단일 액티비티의 데이터 영속화..
            val sharedPref2 = getPreferences(MODE_PRIVATE)
            sharedPref2.edit().run {
                putString("data1", "world")
                putInt("data2", 20)
                apply()
            }
        }

        binding.getBtn.setOnClickListener {
            val sharedPref = getSharedPreferences("my_prefs", MODE_PRIVATE)
            val data1 = sharedPref.getString("data1", "none")
            val data2 = sharedPref.getInt("data2", 0)
            binding.appResult.text = "$data1, $data2"

            val sharedPref2 = getPreferences(MODE_PRIVATE)
            val data3 = sharedPref2.getString("data1", "none")
            val data4 = sharedPref2.getInt("data2", 0)
            binding.activityResult.text = "$data3, $data4"
        }

        binding.goBtn.setOnClickListener {
            val intent = Intent(this, SomeActivity::class.java)
            startActivity(intent)
        }
    }
}